src/snowflake/cli/_plugins/workspace/manager.py

```diff
@@ -31,13 +33,24 @@ def __init__(self, project_definition: ProjectDefinition, project_root: Path):
         self._entities_cache: Dict[str, Entity] = {}
         self._project_definition: DefinitionV20 = project_definition
         self._project_root = project_root
-        self._default_role = default_role()
-        if self._default_role is None:
-            self._default_role = get_sql_executor().current_role()
-        self.default_warehouse = None
-        cli_context = get_cli_context()
-        if cli_context.connection.warehouse:
-            self.default_warehouse = to_identifier(cli_context.connection.warehouse)
+        self._default_role = self._UNSET
+        self._default_warehouse = self._UNSET
+
+    def _get_default_role(self):
```

Comment:
nit: could you add a return type annotation? I think it's important here to label whether the return value is optional or not

Reply:
done